repository.py

```diff
@@ -194,7 +217,11 @@ def __init__(self) -> "Repository":
 
 
 def load_charm(
-    charm: Path, external_libraries: [CharmLibrary], internal_libraries: [CharmLibrary]
+    charm: Path,
+    external_libraries: [CharmLibrary],
+    internal_libraries: [CharmLibrary],
+    binary_packages: dict[str, str],
+    uv_lock: dict[str, Any],
```

Comment:
[nit]: generally it's preferred to use `Mapping` for annotation function arguments rather than `dict` explicitly. It makes it easier to pass objects that inherit from `Mapping` to functions. [Here's relevant bit of Python documentation that I'm referencing](https://docs.python.org/3/library/typing.html#typing.Dict):

```suggestion
    binary_packages: Mapping[str, str],
    uv_lock: Mapping[str, Any],
```

_Note that you'll need to import `Mapping` from `collections.abc`_.